### PR TITLE
feat: edit recovery flow

### LINKED ISSUE #2761: [Recovery] Edit module flow
## What is the feature about

Edit recovery module flow.

## The list of requirements

- [ ] Add edit section
- [ ] Adapt enable flow to allow for editing

## Designs/sketches

- [Figma](https://www.figma.com/file/pO43pjVwR34Uy1ihMsY3OM/Generic-recovery?type=design&node-id=1092-16631&mode=design&t=qqaPfESWw9hO8SKA-0)

## Links

- [Feature spec.](https://www.notion.so/safe-global/Recovery-feature-spec-d873c33a27c24cfabc91c85dbcf3ea35)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #2761

## How this PR fixes it

This modifies the `EditRecoveryFlow` to be `UpsertRecoveryFlow` which now handles editing a recovery Delay Modifier.

## How to test it

Open a Safe with a deployed recovery Delay Modifier and navigate to the (note: temporary) settings, observing the edit button that successfully transactions/alters the settings.

## Checklist
* [x] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/settings/Recovery/DelayModifierRow.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const isOwner = useIsSafeOwner()
+  const [confirm, setConfirm] = useState(false)
+
+  if (!isOwner) {
+    return null
+  }
+
+  const onEdit = () => {
+    // TODO: Display flow
+    setTxFlow(undefined)
```
Comment (schmanu) [inline src/components/settings/Recovery/DelayModifierRow.tsx:24]: I think you forgot to add the UpsertRecoveryFlow here?

File: src/components/settings/Recovery/index.tsx [hunk 2]
```diff
@@ -68,13 +67,16 @@ const headCells = [
   { id: HeadCells.Actions, label: '', sticky: true },
 ]
 
+// TODO: Migrate section
```
Comment (schmanu) [inline src/components/settings/Recovery/index.tsx:70]: Is this still valid?

File: src/components/tx-flow/flows/UpsertRecovery/index.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import type { RecoveryState } from '@/store/recoverySlice'
+
+const Subtitles = ['How does recovery work?', 'Set up recovery settings', 'Set up account recovery']
+
+export enum UpsertRecoveryFlowFields {
+  guardian = 'guardian',
+  txCooldown = 'txCooldown',
+  txExpiration = 'txExpiration',
+  emailAddress = 'emailAddress',
+}
```
Comment (katspaugh) [inline src/components/tx-flow/flows/UpsertRecovery/index.tsx:19]: Are these field names arbitrary or they actually map to ABI params?

File: src/services/recovery/setup.ts [hunk 4]
```diff
@@ -82,3 +88,117 @@ export function getRecoverySetup({
...
+  const guardiansToRemove = _guardians.filter(
+    (oldGuardian) => !newGuardians.some((newGuardian) => sameAddress(newGuardian, oldGuardian)),
+  )
+
+  for (const guardianToAdd of guardiansToAdd) {
+    const enableModule = delayModifierContract.interface.encodeFunctionData('enableModule', [guardianToAdd])
+    txData.push(enableModule)
+
+    // Add guardian to cache
+    _guardians.push(guardianToAdd)
```
Comment (schmanu) [inline src/services/recovery/setup.ts:144]: Do we actually need to add these to the cache?
The guardiansToAdd addresses can not be part of the guardiansToRemove. So they will never be relevant for removing a guardian.

File: src/components/tx-flow/flows/UpsertRecovery/index.tsx (hunk 3 above)
Comment (iamacook) [reply src/components/tx-flow/flows/UpsertRecovery/index.tsx:19]: All but `guardian` map to the ABI.

Comment (katspaugh) [reply src/components/tx-flow/flows/UpsertRecovery/index.tsx:19]: It might make sense to move that to services/recovery then.

Comment (iamacook) [reply src/components/tx-flow/flows/UpsertRecovery/index.tsx:19]: These are only relevant for the fields in this flow though.

File: src/components/settings/Recovery/DelayModifierRow.tsx (hunk 1 above)
Comment (iamacook) [reply src/components/settings/Recovery/DelayModifierRow.tsx]: Indeed, good catch. I've added it in 09cad6f8bc952f8764655756af37d80a1eb86109.

File: src/components/settings/Recovery/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/settings/Recovery/index.tsx]: Yes, the settings need to be merged with spending limits. I've clarified the comment in 09cad6f8bc952f8764655756af37d80a1eb86109.

File: src/services/recovery/setup.ts (hunk 4 above)
Comment (iamacook) [reply src/services/recovery/setup.ts]: True, I've removed it in 09cad6f8bc952f8764655756af37d80a1eb86109.

**Review by katspaugh — APPROVED** [review]

**Pull request merged**